spyder/plugins/remoteclient/widgets/connectiondialog.py

```diff
@@ -613,19 +782,26 @@ def get_icon(self):
     # ---- Public API
     # -------------------------------------------------------------------------
     def save_server_info(self):
-        # Mapping from options in our config system to those accepted by
-        # asyncssh
-        options = SSHClientOptions(
-            host=self.get_option(
-                f"{self.host_id}/{self.auth_method()}/address"
-            ),
-            port=self.get_option(f"{self.host_id}/{self.auth_method()}/port"),
-            username=self.get_option(
-                f"{self.host_id}/{self.auth_method()}/username"
-            ),
-            client_keys=self.get_option(f"{self.host_id}/keyfile"),
-            config=self.get_option(f"{self.host_id}/configfile"),
-        )
+        if self.client_type == ClientType.SSH:
+            # Mapping from options in our config system to those accepted by
+            # asyncssh
+            options = SSHClientOptions(
+                host=self.get_option(
+                    f"{self.host_id}/{self.auth_method()}/address"
+                ),
+                port=self.get_option(f"{self.host_id}/{self.auth_method()}/port"),
+                username=self.get_option(
+                    f"{self.host_id}/{self.auth_method()}/username"
+                ),
+                client_keys=self.get_option(f"{self.host_id}/keyfile"),
+                config=self.get_option(f"{self.host_id}/configfile"),
+            )
+        elif self.client_type == ClientType.JupyterHub:
+            # Mapping from options in our config system to those needed for
+            # connection with a JupyterHub server instance
```

Comment:
```suggestion
            # Mapping from options in our config system to those needed to
            # connect with a JupyterHub server
```

Improve comment a bit.